app/models/competition.rb

```diff
@@ -469,7 +469,11 @@ def warnings_for(user)
       end
 
       if self.results.any? && !self.results_posted?
-        warnings[:results] = I18n.t('competitions.messages.results_not_posted')
+        if user.can_admin_results?
```

Comment:
cursory look suggests that this should work, although it's causing a test to fail. I can try debug it in the next day or so